### DISCUSSION TITLE
compiling grblhal with rgb lights

### DISCUSSION THREAD

**Discussion by mintracer** [grblHAL/core#541] [General]
Hei Terje...

in the past i was able to compile grblhal quite easy... also with the rgb plugin after integrating some lines etc...

since the heavy integration of rgb and neopixels to the core itself, i´m stuggeling with the compiling... the webbuild works fine... 

I would like to change some lights in the code... therefore i need to recompile...

maybe you can give a hint how to do it... it´s not urgent... :)

thanks and a nice week to you!

cheers Einar

**Comment by terjeio** [comment]
which driver/board combo?
with three aux outputs controlling the lights?

**Reply by mintracer** [reply]
i´m running a grblhal2k-EST from Expatria... 
it´s actually flashed with a webbuilder firmware... 
i have a standard RGB cable... so only 3 outputs...

**Reply by terjeio** [reply]
I have forked and [modified](https://github.com/terjeio/grblhal-rgb-plugin) the rgb (status lights) code to work with the new RGB API when available and resort to claiming aux pins when not available (a PR for the changes are [pending](https://github.com/5ocworkshop/grblhal-rgb-plugin/pull/6)). Are you compiling with this version?

**Reply by mintracer** [reply]
i setup a new compile folder, pulled all folders new, inserted the changed rgb plugin from your link, made all the changes according the text on the plugin page additional made the settings in the mymachine.h and i get this error...

![image](https://github.com/grblHAL/core/assets/97637453/5e71b759-1104-440b-8465-cbb1631e6c79)

**Reply by terjeio** [reply]
Seems I have made further changes after the PR, [I just committed them](https://github.com/terjeio/grblhal-rgb-plugin/commit/6e76e03d9a8ee8839451fc29106e7ce224eb6e54). I have renamed `rgb_init()` in the plugin to `status_light_init()` so _grbl/plugins_init.h_ should not be changed. To enable the plugin add `#define STATUS_LIGHT_ENABLE 1` in _my_machine.h_.

Hope this helps.

**Reply by mintracer** [reply]
the build went fine... will test it later on the board!

![image](https://github.com/grblHAL/core/assets/97637453/d2b2a467-e566-4a1f-8a1a-8dea6af0d20b)

Thanks again for the fast response to this "not so important" thing... :) amazing!
